core/src/impl/Kokkos_Error.hpp

```diff
@@ -67,6 +67,50 @@ namespace Impl {
 
 [[noreturn]] void host_abort(const char *const);
 
+#if defined(KOKKOS_ENABLE_CUDA) && defined(__CUDA_ARCH__)
+
+#if defined(__APPLE__) || defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)
+// cuda_abort does not abort when building for macOS.
+// required to workaround failures in random number generator unit tests with
+// pre-volta architectures
+#define KOKKOS_IMPL_ABORT_NORETURN
+#else
+// cuda_abort aborts when building for other platforms than macOS
+#define KOKKOS_IMPL_ABORT_NORETURN [[noreturn]]
+#endif
+
+#elif defined(KOKKOS_ENABLE_HIP) && defined(__HIP_DEVICE_COMPILE__)
+// HIP aborts
+#define KOKKOS_IMPL_ABORT_NORETURN [[noreturn]]
+#elif defined(KOKKOS_ENABLE_SYCL)
```

Comment:
This is the only line I changed (dropped the `&& defined(__SYCL_DEVICE_ONLY__)` to silent warnings).
Admittedly the block could be cleaned up.

Reply:
`Kokkos::abort` called on the `host` should still abort. What kind of warnings were you seeing?

Reply:
You can try locally.  We were getting a warning and we compiler with `-Werror`

Reply:
```diff
diff --git a/core/src/impl/Kokkos_Error.hpp b/core/src/impl/Kokkos_Error.hpp
index 85a434739..56c507785 100644
--- a/core/src/impl/Kokkos_Error.hpp
+++ b/core/src/impl/Kokkos_Error.hpp
@@ -82,7 +82,7 @@ namespace Impl {
 #elif defined(KOKKOS_ENABLE_HIP) && defined(__HIP_DEVICE_COMPILE__)
 // HIP aborts
 #define KOKKOS_IMPL_ABORT_NORETURN [[noreturn]]
-#elif defined(KOKKOS_ENABLE_SYCL)
+#elif defined(KOKKOS_ENABLE_SYCL) && defined(__SYCL_DEVICE_ONLY__)
 // FIXME_SYCL SYCL doesn't abort
 #define KOKKOS_IMPL_ABORT_NORETURN
 #elif !defined(KOKKOS_ENABLE_OPENMPTARGET)
@@ -93,9 +93,16 @@ namespace Impl {
 #define KOKKOS_IMPL_ABORT_NORETURN
 #endif
 
+// FIXME_SYCL
+#ifdef KOKKOS_ENABLE_SYCL
+#define KOKKOS_IMPL_ABORT_NORETURN_DEVICE
+#else
+#define KOKKOS_IMPL_ABORT_NORETURN_DEVICE KOKKOS_IMPL_ABORT_NORETURN
+#endif
+
 #if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
     defined(KOKKOS_ENABLE_SYCL) || defined(KOKKOS_ENABLE_OPENMPTARGET)
-KOKKOS_IMPL_ABORT_NORETURN inline KOKKOS_IMPL_DEVICE_FUNCTION void device_abort(
+KOKKOS_IMPL_ABORT_NORETURN_DEVICE inline KOKKOS_IMPL_DEVICE_FUNCTION void device_abort(
     const char *const msg) {
 #if defined(KOKKOS_ENABLE_CUDA)
   ::Kokkos::Impl::cuda_abort(msg);
```
fixes the warning for me.

Reply:
I can apply, not going to fight for this, but you realize how convoluted this is getting just to slap a `[[noreturn]]` attribute on it on the host-side only.

Reply:
Yeah, I don't think it's that important.